tests/integration/test_storage3.py

```diff
@@ -97,88 +99,89 @@ def test_list_buckets():
     assert bucket_list[0]["Name"] == TEST_BUCKET
 
 
-@pytest.mark.usefixtures("create_large_object")
-def test_create_multipart_upload():
-    config = get_config()
-    manager = Boto3Manager(config)
-    src_url = f"s3://localhost:7000/{TEST_BUCKET}/{ORIGINAL_FILE_NAME}"
-    dst_url = f"s3://localhost:7000/{TEST_BUCKET}/{COPIED_FILE_NAME}"
-    mp_info = manager.create_multipart_upload(src_url=src_url, dst_url=dst_url)
-
-    assert "md5_sum" in mp_info
-    mp_info.pop("md5_sum")
-    assert "sha256_sum" in mp_info
-    mp_info.pop("sha256_sum")
-    assert "mp_id" in mp_info
-    mp_info.pop("mp_id")
-    assert "stream_buffer" in mp_info
-    mp_info.pop("stream_buffer")
-    assert "start_time" in mp_info
-    mp_info.pop("start_time")
-
-    assert mp_info == {
-        "dst_info": {
-            "url": "s3://localhost:7000/test_bucket/copied_file",
-            "s3_loc": "localhost:7000",
-            "bucket_name": "test_bucket",
-            "key_name": "copied_file",
+def moto_server_gen(hostname: str, port: int, is_secure: bool):
+    mock = MotoServer(hostname=hostname, port=port, is_secure=is_secure)
+    mock.start()
+    yield mock
+    mock.stop()
+
+
+def gen_moto_server(hostname="localhost", port=7001, is_secure=True):
+    mock = MotoServer(hostname=hostname, port=port, is_secure=is_secure)
+    mock.start()
+    yield mock
+    mock.stop()
+
+
+def test_simulate_cleversafe_to_aws_multipart_copy(create_large_file):
+    """
+    The multipart upload is used to support transferring large files from one S3 provider to another.
+    """
+    # Start the servers
+    host_a = "localhost"
+    port_a = 7001
+    url_a = f"{host_a}:{port_a}"
+    moto_server_a = next(gen_moto_server(hostname=host_a, port=port_a))
```

Comment:
This should be handled how @wrkhenddher is suggesting. Otherwise these will never get closed. You'd have to call next again on the generator for that to happen.